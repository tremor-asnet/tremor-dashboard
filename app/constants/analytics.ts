/* Constants for Analytics Page */
export const SALE_STATISTICAL_TYPE = {
  REVENUE: "Revenue",
};
