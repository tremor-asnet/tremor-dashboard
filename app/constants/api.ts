import { ROUTER_API_URL } from "./routes";

export const API_ROUTES = {
  USERS: `${ROUTER_API_URL}/users`,
};
