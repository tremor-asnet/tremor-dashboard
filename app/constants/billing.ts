export const enum AGGREGATION_TYPE {
  SALARY = "Salary",
  PAYPAL = "Paypal",
}

export const enum AGGREGATION_DESCRIPTION {
  SALARY = "Belong Interactive",
  PAYPAL = "Freelance Payment",
}
