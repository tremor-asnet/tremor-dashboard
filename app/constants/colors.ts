type ColorChannelsChartType = {
  [key: number]: string;
  1: string;
  2: string;
  3: string;
  4: string;
};

export const CHANNELS_CHART_COLOR: ColorChannelsChartType = {
  1: "blue-500",
  2: "red-500",
  3: "primary",
  4: "gray-500",
};

export const TRANSACTION_COLOR = {
  INCREASE: "few",
  DECREASE: "attention",
  PENDING: "secondary",
};
