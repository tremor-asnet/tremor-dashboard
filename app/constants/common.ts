export const EXCEPT_KEYS = {
  POSITIVE_INTEGER: ["e", "E", "+", "-", "."],
  POSITIVE_DOUBLE: ["e", "E", "+", "-"],
};

export const DIRECTION = {
  ASC: "asc",
  DESC: "desc",
};
