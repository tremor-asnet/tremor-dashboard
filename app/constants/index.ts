export * from "./routes";
export * from "./regex";
export * from "./messages";
export * from "./profile";
export * from "./commons";
export * from "./flagSrc";
export * from "./images";
export * from "./sections";
export * from "./images";
export * from "./products";
export * from "./colors";
export * from "./units";
export * from "./sales";
export * from "./commons";
export * from "./status";
