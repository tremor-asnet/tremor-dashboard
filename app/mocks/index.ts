export * from "./card";
export * from "./profile";
export * from "./project";
export * from "./charts";
export * from "./analytics";
export * from "./sales";
export * from "./table-list";
export * from "./orders";
export * from "./product";
