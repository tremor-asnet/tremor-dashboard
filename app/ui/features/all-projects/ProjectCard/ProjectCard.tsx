"use client";

//Libs
import { useState, RefObject } from "react";
import { Card, Text, Flex, Title, Button } from "@tremor/react";
import { FaEllipsisV } from "react-icons/fa";

//Components
import { CustomImage, Avatar } from "@/ui/components";

//Types
import { Project, AvatarCard } from "@/types";

//Constants
import {
  ITEM_ACTION_PROJECT,
  PLACEHOLDER_IMAGE,
  VARIANT_BUTTON,
} from "@/constants";

//Mocks
import { PROJECT_DATA } from "@/mocks/project";

//Helpers
import { formatDate } from "@/helpers";

// Hooks
import { useOutsideClick } from "@/hooks";

interface ActionCardProps {
  key: string;
  label: string;
}
interface ProjectCardProps {
  projectData: Project;
  actions: ActionCardProps[];
}

const ProjectCard = ({
  projectData = PROJECT_DATA[0],
  actions = ITEM_ACTION_PROJECT,
}: ProjectCardProps): JSX.Element => {
  const { cover, name, dueDate, participants, description, id } = projectData;
  const participantNumber = participants?.length;
  const duaDateFormat = formatDate(new Date(dueDate));
  const srcImageCard = cover ? cover : PLACEHOLDER_IMAGE;

  const [isOpenAction, setOpenAction] = useState(false);
  const [currentProjectId, setCurrentProjectId] = useState("");
  const openActionProject = isOpenAction && id === currentProjectId;

  const projectCardRef = useOutsideClick(() => {
    setOpenAction(false);
  });

  const handleItemActionProject = () => {
    setOpenAction(false);
  };

  const handleToggleAction = (id: string) => {
    setOpenAction(!isOpenAction);
    setCurrentProjectId(id);
  };

  return (
    <div className="antialiased items-center justify-between pb-1">
      <div className="flex items-center">
        <Card
          className="dark:bg-dark-tremor-primary mx-auto px-4 py-1 ring-0 max-w-full lg:max-w-[356px] 2xl:max-w-full border-none relative mt-[40px] rounded-xl shadow-md"
          ref={projectCardRef as RefObject<HTMLDivElement>}>
          <Flex className="absolute top-[-22px] left-40px w-[74px] h-[74px] p-1 bg-gradient-arsenic justify-center rounded-xl dark:bg-gradient-pickled">
            <CustomImage
              src={srcImageCard}
              width={60}
              height={60}
              alt={`${name}-cover`}
              priority
              className="rounded-md"
            />
          </Flex>
          <Flex className="pl-[90px] mb-6 mt-1 relative">
            <Flex flexDirection="col" alignItems="start" justifyContent="start">
              <Title className="text-xl text-tremor-content-title dark:text-dark-tremor-content-title font-semibold text-base leading-5">
                {name}
              </Title>
              <Flex
                justifyContent="start"
                alignItems="start"
                className="mt-1 ml-[10px]">
                {participants?.map((participant: AvatarCard, index: number) => (
                  <Avatar
                    key={`Avatar ${index} of ${name} by ${id}`}
                    alt="Avatar extra small"
                    className="border-2 border-white border-solid ml-[-10px] w-6 h-6 min-w-[24px]"
                    height={20}
                    priority
                    src={participant?.avatar}
                    sizes="(max-width: 768px) 100vw, 33vw"
                    width={20}
                  />
                ))}
              </Flex>
            </Flex>
            <Flex flexDirection="col" justifyContent="end" className="w-auto">
              <Flex
                flexDirection="col"
                className="cursor-pointer w-[30px] h-[16px]"
                onClick={() => handleToggleAction(projectData.id)}>
                <FaEllipsisV className="text-secondary dark:text-secondary" />
              </Flex>
              {openActionProject && (
                <div className="absolute p-2 right-[26px] top-[3px] z-10 bg-white dark:bg-dark-tremor-primary rounded-md shadow-md dark:shadow-select-option">
                  {actions.map(item => (
                    <Flex key={item.key}>
                      <Button
                        className="w-full justify-start text-tremor-content-title hover:text-tremor-content-title hover:bg-body dark:hover:bg-dark-secondary hover:rounded-md py-[6px]"
                        variant={VARIANT_BUTTON.LIGHT}
                        onClick={() => handleItemActionProject()}>
                        <Text className="px-3 font-normal text-sm text-secondary dark:text-lighter hover:text-primary dark:hover:text-white">
                          {item.label}
                        </Text>
                      </Button>
                    </Flex>
                  ))}
                </div>
              )}
            </Flex>
          </Flex>
          <Text className="text-md text-secondary my-4 font-light dark:text-dark-tremor-content-romance">
            {description}
          </Text>
          <div className="h-px bg-[linear-gradient(to_right,rgba(52,71,103,0),rgba(52,71,103,0.4),rgba(52,71,103,0))] dark:bg-gradient-divider opacity-25 my-4" />
          <Flex className="my-4">
            <div>
              <Text className="text-primary font-semibold dark:text-dark-tremor-content-title">
                {participantNumber}
              </Text>
              <Text className="text-secondary dark:text-secondary font-normal">
                Participants
              </Text>
            </div>
            <div>
              <Text className="text-primary font-semibold dark:text-dark-tremor-content-title">
                {duaDateFormat}
              </Text>
              <Text className="text-secondary dark:text-secondary font-normal">
                Due date
              </Text>
            </div>
          </Flex>
        </Card>
      </div>
    </div>
  );
};

export default ProjectCard;
