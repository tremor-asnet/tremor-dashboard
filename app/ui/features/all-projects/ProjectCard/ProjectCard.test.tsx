import { fireEvent, render, waitFor } from "@testing-library/react";

//Components
import ProjectCard from "./ProjectCard";

//Constans
import { ITEM_ACTION_PROJECT } from "@/constants";

//Mocks
import { PROJECT_DATA } from "@/mocks/project";

describe("Testing ProjectCard component", () => {
  const propsDefault = {
    projectData: PROJECT_DATA[0],
    actions: ITEM_ACTION_PROJECT,
  };

  const renderWrapper = () => render(<ProjectCard {...propsDefault} />);

  it("Should match snapshot", () => {
    const component = renderWrapper();

    expect(component).toMatchSnapshot();
  });

  it("Should toggle action menu when ellipsis icon is clicked", async () => {
    const { findByTestId } = renderWrapper();

    fireEvent.click(await findByTestId("toggle-icon"));

    expect(await findByTestId("menu-action")).toBeTruthy();
  });

  it("Should close action menu when clicking outside the component", async () => {
    const renderWrapperWithOutSide = () =>
      render(
        <div data-testid="outside">
          <ProjectCard {...propsDefault} />
        </div>,
      );
    const { findByTestId } = renderWrapperWithOutSide();

    fireEvent.click(await findByTestId("toggle-icon"));
    fireEvent.click(await findByTestId("outside"));

    waitFor(() => expect(findByTestId("menu-action")).toBeFalsy());
  });

  it("Should close action menu when an action is clicked", async () => {
    const { findByTestId } = renderWrapper();

    fireEvent.click(await findByTestId("toggle-icon"));
    fireEvent.click(await findByTestId(ITEM_ACTION_PROJECT[0].key));

    waitFor(() => expect(findByTestId("menu-action")).toBeFalsy());
  });
});
