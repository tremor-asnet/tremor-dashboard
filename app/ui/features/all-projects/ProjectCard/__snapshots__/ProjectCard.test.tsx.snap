// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ProjectCard component Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="antialiased items-center justify-between pb-1"
      >
        <div
          class="flex items-center"
        >
          <div
            class="tremor-Card-root w-full text-left rounded-tremor-default bg-tremor-background ring-tremor-ring shadow-tremor-card dark:ring-dark-tremor-ring dark:shadow-dark-tremor-card border-tremor-brand dark:border-dark-tremor-brand p-6 dark:bg-dark-tremor-primary mx-auto px-4 py-1 ring-0 max-w-full lg:max-w-[356px] 2xl:max-w-full border-none relative mt-[40px] rounded-xl shadow-md"
          >
            <div
              class="tremor-Flex-root flex flex-row items-center absolute top-[-22px] left-40px w-[74px] h-[74px] p-1 bg-gradient-arsenic justify-center rounded-xl dark:bg-gradient-pickled"
            >
              <img
                alt="Slack Bot-cover"
                class="rounded-md"
                data-nimg="1"
                decoding="async"
                fetchpriority="high"
                height="60"
                src="https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/logo-slack.0aa818de.svg"
                style="color: transparent;"
                width="60"
              />
            </div>
            <div
              class="tremor-Flex-root flex w-full flex-row justify-between items-center pl-[90px] mb-6 mt-1 relative"
            >
              <div
                class="tremor-Flex-root flex w-full flex-col justify-start items-start"
              >
                <p
                  class="text-tremor-title text-tremor-content-title dark:text-dark-tremor-content-title font-semibold text-base leading-5"
                >
                  Slack Bot
                </p>
                <div
                  class="tremor-Flex-root flex w-full flex-row justify-start items-start mt-1 ml-[10px]"
                >
                  <div
                    class="inline-flex rounded-full overflow-hidden border-2 border-white border-solid ml-[-10px] w-6 h-6 min-w-[24px]"
                  >
                    <img
                      alt="Avatar extra small"
                      data-nimg="1"
                      decoding="async"
                      fetchpriority="high"
                      height="20"
                      sizes="(max-width: 768px) 100vw, 33vw"
                      src="http://localhost/_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=3840&q=75"
                      srcset="/_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=256&q=75 256w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=384&q=75 384w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=640&q=75 640w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=750&q=75 750w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=828&q=75 828w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=3840&q=75 3840w"
                      style="color: transparent;"
                      width="20"
                    />
                  </div>
                  <div
                    class="inline-flex rounded-full overflow-hidden border-2 border-white border-solid ml-[-10px] w-6 h-6 min-w-[24px]"
                  >
                    <img
                      alt="Avatar extra small"
                      data-nimg="1"
                      decoding="async"
                      fetchpriority="high"
                      height="20"
                      sizes="(max-width: 768px) 100vw, 33vw"
                      src="http://localhost/_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=3840&q=75"
                      srcset="/_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=256&q=75 256w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=384&q=75 384w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=640&q=75 640w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=750&q=75 750w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=828&q=75 828w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=3840&q=75 3840w"
                      style="color: transparent;"
                      width="20"
                    />
                  </div>
                  <div
                    class="inline-flex rounded-full overflow-hidden border-2 border-white border-solid ml-[-10px] w-6 h-6 min-w-[24px]"
                  >
                    <img
                      alt="Avatar extra small"
                      data-nimg="1"
                      decoding="async"
                      fetchpriority="high"
                      height="20"
                      sizes="(max-width: 768px) 100vw, 33vw"
                      src="http://localhost/_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=3840&q=75"
                      srcset="/_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=256&q=75 256w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=384&q=75 384w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=640&q=75 640w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=750&q=75 750w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=828&q=75 828w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=3840&q=75 3840w"
                      style="color: transparent;"
                      width="20"
                    />
                  </div>
                </div>
              </div>
              <div
                class="tremor-Flex-root flex flex-col justify-end items-center w-auto"
              >
                <div
                  class="tremor-Flex-root flex flex-col justify-between items-center cursor-pointer w-[30px] h-[16px]"
                  data-testid="toggle-icon"
                >
                  <svg
                    class="text-secondary dark:text-secondary"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 192 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <p
              class="text-tremor-default text-md text-secondary my-4 font-light dark:text-dark-tremor-content-romance"
            >
              If everything I did failed - which it doesn't, I think that it actually succeeds.
            </p>
            <div
              class="h-px bg-[linear-gradient(to_right,rgba(52,71,103,0),rgba(52,71,103,0.4),rgba(52,71,103,0))] dark:bg-gradient-divider opacity-25 my-4"
            />
            <div
              class="tremor-Flex-root flex w-full flex-row justify-between items-center my-4"
            >
              <div>
                <p
                  class="text-tremor-default text-primary font-semibold dark:text-dark-tremor-content-title"
                >
                  3
                </p>
                <p
                  class="text-tremor-default text-secondary dark:text-secondary font-normal"
                >
                  Participants
                </p>
              </div>
              <div>
                <p
                  class="text-tremor-default text-primary font-semibold dark:text-dark-tremor-content-title"
                >
                  03.02.22
                </p>
                <p
                  class="text-tremor-default text-secondary dark:text-secondary font-normal"
                >
                  Due date
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="antialiased items-center justify-between pb-1"
    >
      <div
        class="flex items-center"
      >
        <div
          class="tremor-Card-root w-full text-left rounded-tremor-default bg-tremor-background ring-tremor-ring shadow-tremor-card dark:ring-dark-tremor-ring dark:shadow-dark-tremor-card border-tremor-brand dark:border-dark-tremor-brand p-6 dark:bg-dark-tremor-primary mx-auto px-4 py-1 ring-0 max-w-full lg:max-w-[356px] 2xl:max-w-full border-none relative mt-[40px] rounded-xl shadow-md"
        >
          <div
            class="tremor-Flex-root flex flex-row items-center absolute top-[-22px] left-40px w-[74px] h-[74px] p-1 bg-gradient-arsenic justify-center rounded-xl dark:bg-gradient-pickled"
          >
            <img
              alt="Slack Bot-cover"
              class="rounded-md"
              data-nimg="1"
              decoding="async"
              fetchpriority="high"
              height="60"
              src="https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/logo-slack.0aa818de.svg"
              style="color: transparent;"
              width="60"
            />
          </div>
          <div
            class="tremor-Flex-root flex w-full flex-row justify-between items-center pl-[90px] mb-6 mt-1 relative"
          >
            <div
              class="tremor-Flex-root flex w-full flex-col justify-start items-start"
            >
              <p
                class="text-tremor-title text-tremor-content-title dark:text-dark-tremor-content-title font-semibold text-base leading-5"
              >
                Slack Bot
              </p>
              <div
                class="tremor-Flex-root flex w-full flex-row justify-start items-start mt-1 ml-[10px]"
              >
                <div
                  class="inline-flex rounded-full overflow-hidden border-2 border-white border-solid ml-[-10px] w-6 h-6 min-w-[24px]"
                >
                  <img
                    alt="Avatar extra small"
                    data-nimg="1"
                    decoding="async"
                    fetchpriority="high"
                    height="20"
                    sizes="(max-width: 768px) 100vw, 33vw"
                    src="http://localhost/_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=3840&q=75"
                    srcset="/_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=256&q=75 256w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=384&q=75 384w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=640&q=75 640w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=750&q=75 750w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=828&q=75 828w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=3840&q=75 3840w"
                    style="color: transparent;"
                    width="20"
                  />
                </div>
                <div
                  class="inline-flex rounded-full overflow-hidden border-2 border-white border-solid ml-[-10px] w-6 h-6 min-w-[24px]"
                >
                  <img
                    alt="Avatar extra small"
                    data-nimg="1"
                    decoding="async"
                    fetchpriority="high"
                    height="20"
                    sizes="(max-width: 768px) 100vw, 33vw"
                    src="http://localhost/_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=3840&q=75"
                    srcset="/_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=256&q=75 256w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=384&q=75 384w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=640&q=75 640w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=750&q=75 750w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=828&q=75 828w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=3840&q=75 3840w"
                    style="color: transparent;"
                    width="20"
                  />
                </div>
                <div
                  class="inline-flex rounded-full overflow-hidden border-2 border-white border-solid ml-[-10px] w-6 h-6 min-w-[24px]"
                >
                  <img
                    alt="Avatar extra small"
                    data-nimg="1"
                    decoding="async"
                    fetchpriority="high"
                    height="20"
                    sizes="(max-width: 768px) 100vw, 33vw"
                    src="http://localhost/_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=3840&q=75"
                    srcset="/_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=256&q=75 256w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=384&q=75 384w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=640&q=75 640w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=750&q=75 750w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=828&q=75 828w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=3840&q=75 3840w"
                    style="color: transparent;"
                    width="20"
                  />
                </div>
              </div>
            </div>
            <div
              class="tremor-Flex-root flex flex-col justify-end items-center w-auto"
            >
              <div
                class="tremor-Flex-root flex flex-col justify-between items-center cursor-pointer w-[30px] h-[16px]"
                data-testid="toggle-icon"
              >
                <svg
                  class="text-secondary dark:text-secondary"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 192 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <p
            class="text-tremor-default text-md text-secondary my-4 font-light dark:text-dark-tremor-content-romance"
          >
            If everything I did failed - which it doesn't, I think that it actually succeeds.
          </p>
          <div
            class="h-px bg-[linear-gradient(to_right,rgba(52,71,103,0),rgba(52,71,103,0.4),rgba(52,71,103,0))] dark:bg-gradient-divider opacity-25 my-4"
          />
          <div
            class="tremor-Flex-root flex w-full flex-row justify-between items-center my-4"
          >
            <div>
              <p
                class="text-tremor-default text-primary font-semibold dark:text-dark-tremor-content-title"
              >
                3
              </p>
              <p
                class="text-tremor-default text-secondary dark:text-secondary font-normal"
              >
                Participants
              </p>
            </div>
            <div>
              <p
                class="text-tremor-default text-primary font-semibold dark:text-dark-tremor-content-title"
              >
                03.02.22
              </p>
              <p
                class="text-tremor-default text-secondary dark:text-secondary font-normal"
              >
                Due date
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
